### ISSUE TITLE
Incorrect return type at getListByCustomerId in PaymentTokenManagementInterface

### ISSUE THREAD

**Issue by eduan** [magento/magento2#22899]
### Summary (*)
Magento 2.2.7
Magento 2.3.0 
I'm trying to release the PaymentTokenManagement ->getListByCustomerId via API, but I'm facing the following issue:

``` 
[2019-05-14 19:00:20] main.CRITICAL: Report ID: webapi-5cdb104406787; Message: Class \Magento\Vault\Api\Data\PaymentTokenSearchResultsInterfa does not exist {"exception":"[object] (Exception(code: -1): Report ID: webapi-5cdb104406787; Message: Class \\Magento\\Vault\\Api\\Data\\PaymentTokenSearchResultsInterfa does not exist at /var/www/magento22/lib/internal/Magento/Framework/Webapi/ErrorProcessor.php:205, ReflectionException(code: -1): Class \\Magento\\Vault\\Api\\Data\\PaymentTokenSearchResultsInterfa does not exist at /var/www/magento22/lib/internal/Magento/Framework/Reflection/MethodsMap.php:155)"} []
```
it's due to the fact that the Interface return type documented is different from its implementation.

### Examples (*)

Interface method documentation (app/code/Magento/Vault/Api/PaymentTokenManagementInterface.php):    

``` 
/**
     * Lists payment tokens that match specified search criteria.
     *
     * @param int $customerId Customer ID.
     * @return \Magento\Vault\Api\Data\PaymentTokenSearchResultsInterface Payment token search result interface.
     * @since 100.1.0
     */
``` 

Implementation documentation (app/code/Magento/Vault/Model/PaymentTokenManagement.php)
``` 
/**
     * Lists payment tokens that match specified search criteria.
     *
     * @param int $customerId Customer ID.
     * @return \Magento\Vault\Api\Data\PaymentTokenSearchResultsInterface[] Payment tokens search result interface.
     */
``` 
### Proposed solution

As the API result is interpreted using the annotations. My proposal is to change the interface return type to `\Magento\Vault\Api\Data\PaymentTokenSearchResultsInterface[]`, because it's actually returning a array of PaymentTokenSearchResultsInterface .

**Comment by irajneeshgupta** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta, here is your [Magento instance](https://i-22899-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-22899-2-2-develop.instances.magento-community.engineering/admin](https://i-22899-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by irajneeshgupta** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta, here is your [Magento instance](https://i-22899-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22899-2-3-develop.instances.magento-community.engineering/admin](https://i-22899-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @krishprakash 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99712`, `MAGETWO-99713` were created

**Issue Available**: @krishprakash, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @eduan. Thank you for your report. 
The issue has been fixed in magento/magento2#22914 by @serhiyzhovnir in 2.3-develop branch
Related commit(s):
 - [6e5cdb1bd969bee2ab2cd5de89e4bf9558348552](https://github.com/magento/magento2/commit/6e5cdb1bd969bee2ab2cd5de89e4bf9558348552)
 - [f1c85d4cc12328412e45ef5d51012798513eb0b2](https://github.com/magento/magento2/commit/f1c85d4cc12328412e45ef5d51012798513eb0b2)

The fix will be available with the upcoming 2.3.3 release.

**Comment by magento-engcom-team** [comment]
Hi @eduan. Thank you for your report. 
The issue has been fixed in magento/magento2#22915 by @serhiyzhovnir in 2.2-develop branch
Related commit(s):
 - [2f645535374757693ed7699e4aa67a830f00fedd](https://github.com/magento/magento2/commit/2f645535374757693ed7699e4aa67a830f00fedd)
 - [f3ef75404259361a25e92b4afd225fd2960c8249](https://github.com/magento/magento2/commit/f3ef75404259361a25e92b4afd225fd2960c8249)

The fix will be available with the upcoming 2.2.10 release.
